### PR TITLE
fix(video): fix video resolution setting

### LINKED ISSUE #1033: Video Resolution setting not retained
I'm setting Video Resolution to 160x120, but if I move out of the Audio/Video tab and back, the resolution is back to the highest. This is as of commit aef083934f8a9366a855ec474bd6dec7f3f88867

### PR DESCRIPTION
**Pull request by b4n**
Index the filtered modes we show to the user when selecting from the
combobox, not the full list of modes actually supported by the device.

Fixes #1033.

### CODE REVIEW

File: src/widget/form/settings/avform.cpp [hunk 1]
```diff
@@ -162,8 +162,8 @@ void AVForm::updateVideoModes(int curIndex)
         return;
     }
     QString devName = videoDeviceList[curIndex].first;
-    videoModes = CameraDevice::getVideoModes(devName);
-    std::sort(videoModes.begin(), videoModes.end(),
+    QVector<VideoMode> allVideoModes = CameraDevice::getVideoModes(devName);
```
Comment (b4n) [inline src/widget/form/settings/avform.cpp:165]: A large part of the diff relates to renaming this variable.

File: src/widget/form/settings/avform.cpp [hunk 2]
```diff
@@ -227,12 +227,14 @@ void AVForm::updateVideoModes(int curIndex)
     QSize prefRes = Settings::getInstance().getCamVideoRes();
     unsigned short prefFPS = Settings::getInstance().getCamVideoFPS();
     // Iterate backwards to show higest resolution first.
```
Comment (b4n) [inline src/widget/form/settings/avform.cpp:229]: The actually interesting changes start here

File: src/widget/form/settings/avform.cpp [hunk 3]
```diff
@@ -227,12 +227,14 @@ void AVForm::updateVideoModes(int curIndex)
...
+    videoModes.clear();
     for(auto iter = bestModeInds.rbegin(); iter != bestModeInds.rend(); ++iter)
     {
         int i = iter->second;
-        VideoMode mode = videoModes[i];
+        VideoMode mode = allVideoModes[i];
+        videoModes.push_back(mode);
```
Comment (sudden6) [inline src/widget/form/settings/avform.cpp]: Since we use qt it would be better to use `append(mode);` here.

Comment (b4n) [inline src/widget/form/settings/avform.cpp]: I wondered for a second, grepped the source and found it was more or less random apparently, so I kept what I had, but I'll change it.

``` console
$ git grep -F .append | wc -l
42
$ git grep -F .push_back | wc -l
33
```

**Comment by b4n** [comment]
@sudden6 I updated the patch.

**Pull request merged**